src/TestProblem/ELBDM/HaloMerger/Init_TestProb_ELBDM_HaloMerger.cpp

```diff
@@ -0,0 +1,1282 @@
+#include "GAMER.h"
+#include "TestProb.h"
+
+
+
+// problem-specific global variables
+// =======================================================================================
+static int      HaloMerger_Halo_Num;                                       // total number of halos [2]
+static int      HaloMerger_Halo_InitMode;                                  // halo initialization mode [1]
+                                                                           //   (1=single-level UM_IC of real and imaginary parts)
+static int      HaloMerger_Soliton_Num;                                    // total number of solitons [0]
+static int      HaloMerger_Soliton_InitMode;                               // soliton initialization mode [1]
+                                                                           //   (1=table of the density profile, 2=analytical function of the density profile)
+
+// External Potential-related parameters to read from the input
+       double   HaloMerger_ExtPot_UniDenSph_M;                             // mass of the uniform-density sphere for the external potential (must >= 0.0) [0.0]
+       double   HaloMerger_ExtPot_UniDenSph_R;                             // radius of the uniform-density sphere for the external potential (must > 0.0) [-1.0]
+       double   HaloMerger_ExtPot_UniDenSph_CenCoordX;                     // x-coordinate of the center of the uniform-density sphere for the external potential (<0.0=auto -> box center) [-1.0]
+       double   HaloMerger_ExtPot_UniDenSph_CenCoordY;                     // y-coordinate of the center of the uniform-density sphere for the external potential (<0.0=auto -> box center) [-1.0]
+       double   HaloMerger_ExtPot_UniDenSph_CenCoordZ;                     // z-coordinate of the center of the uniform-density sphere for the external potential (<0.0=auto -> box center) [-1.0]
+       double   HaloMerger_ExtPot_UniDenSph_VelocityX;                     // x-component of the velocity of the uniform-density sphere for the external potential [0.0]
+       double   HaloMerger_ExtPot_UniDenSph_VelocityY;                     // y-component of the velocity of the uniform-density sphere for the external potential [0.0]
+       double   HaloMerger_ExtPot_UniDenSph_VelocityZ;                     // z-component of the velocity of the uniform-density sphere for the external potential [0.0]
+static double   HaloMerger_ExtPot_UniDenSph_Rho;                           // density of the uniform-density sphere for the external potential
+
```

Comment:
The following halo and soliton `char` variables can be declared as local variables within `SetParameter()`.